doc/developer/platform/formalism.md

```diff
@@ -40,12 +40,12 @@ are faithful to the "real" TVC between `since` and `upper`.
 
 Materialize identifies each TVC by a unique `GlobalId`, and maintains a mutable map of `GlobalId`s to the current pTVC representing that `GlobalID`'s TVC.
 
-Internally, Materialize is separated into two layers: *storage* and *compute*.
-The storage layer records updates from (e.g.) external data sources. The
-compute layer provides materialized views: new TVCs which are derived from
-other TVCs in some way. The ability to write and read to these layers is given
-by a first-class *capability* object, which constrains when pTVCs can be
-compacted and appended to.
+Internally, Materialize is separated into three layers: *storage*, *compute*,
```

Comment:
The adapter is immediately described in the next paragraph so it makes sense to list it here as one of the layers.

Reply:
Meta request: some sentences in this doc are one per line. Others are wrapped. Encouragement to make everything one per line, so that diffs like this are much easier to read.

Reply:
Should I do that as part of this PR or a separate PR? I think that if it's included in this PR, then it will be even harder to read the diffs.

Reply:
Whatever works for you, including just not doing this at all. If it were me, I'd make this PR 2 commits. First commit is re-formatting. Second is the changes here. Then the diffs are useful. I also understand that finessing 2 commits is not common git knowledge.

Reply:
I updated the PR so it doesn't change the line numbers of any existing text. That way the diffs are much easier to read.

I'm not against making sentences one per line, I just don't particularly want to do it as part of this PR.